doc/api/n-api.md

```diff
@@ -2492,6 +2492,35 @@ This API creates a JavaScript `symbol` value from a UTF8-encoded C string.
 The JavaScript `symbol` type is described in [Section 19.4][]
 of the ECMAScript Language Specification.
 
+#### `node_api_symbol_for`
+
+<!-- YAML
+added: REPLACEME
+napiVersion: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+```c
+napi_status node_api_symbol_for(napi_env env,
+                                const char* description,
```

Comment:
We discussed in Node-API team meeting and the team was ok adding this API, but the suggestion was that to be consistent with other APIs, it should take a length (which can be NAPI_AUTO_LENGTH)

Also some tweaks for the parameter names as follows:

napi_status node_api_symbol_for(napi_env env,
                                const char* utf8description,
                                size_t length,
                                napi_value* result);


And the in the implementation use https://github.com/nodejs/node/blob/7899fdb037ae8d3afee8d018da0899d566c30eb7/doc/api/n-api.md#napi_create_string_utf8 to create the JavaScript string (since it already handles the length)

Reply:
@mhdawson thanks, updated it to accept a length.

@addaleax since you had some concerns about null descriptions in https://github.com/nodejs/node/pull/41329#discussion_r775269221, I'd like to point out that the current implementation now accepts null descriptions and converts it into an empty string only if the length is set to 0 like how `napi_create_string_utf8()` works, otherwise it throws `Error: Invalid argument`. Does this look okay to you?

Reply:
Yes :+1: Calling `Symbol.for()` with an empty string is still almost certainly a bug, but I guess it’s not one that we can detect through an API like this without excluding valid use cases.